blocks/fast_reader_test.go

```diff
@@ -36,33 +47,57 @@ func TestFastRead(t *testing.T) {
 	}
 
 	for i, test := range tests {
-		buffer := new(bytes.Buffer)
-		w := NewWriter(buffer, test.blockSize)
-
-		w.Write([]byte(test.input))
-		w.Flush()
-
-		r := NewFastReader(context.Background(), bytes.NewReader(buffer.Bytes()), test.blockSize)
-
-		var err error
-		var length int
-		result := make([]byte, 0)
-		bytes := make([]byte, test.readSize)
-		n := 1
-
-		for n > 0 && err == nil {
-			n, err = r.Read(bytes)
-			result = append(result, bytes[:n]...)
-			length += n
-		}
-
-		if !reflect.DeepEqual(result, []byte(test.input)) {
-			t.Errorf("Wrong bytes for Case %d:\n want: %x\n  got: %x", i, test.input, result)
-		}
-
-		if length != len(test.input) || err != io.EOF {
-			t.Errorf("Wrong return for Case %d: want: %d,EOF got: %d,%v", i, len(test.input), length, err)
-		}
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			buffer := new(bytes.Buffer)
+			w := NewWriter(buffer, test.blockSize)
+
+			w.Write([]byte(test.input))
+			w.Flush()
+
+			// r := NewFastReader(context.Background(), bytes.NewReader(buffer.Bytes()), test.blockSize)
+			f := bytes.NewReader(buffer.Bytes())
+			fr := fileReaderPool.Get().(*bufio.Reader)
+			fr.Reset(f)
+
+			// defer func() {
+			// 	fr.Reset(nil)
+			// 	fileReaderPool.Put(fr)
+			// }()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			r := NewFastReader(ctx, fr, test.blockSize)
+
+			defer func() {
+				cancel()
+				r.Close()
+				fr.Reset(nil)
+				fileReaderPool.Put(fr)
+			}()
+
+			if true {
+				return
+			}
```

Comment:
This is here just to trigger the defer. I also updated the test to use `t.Run` in each iteration, so that the defer would run for each test case.